### DISCUSSION TITLE
trivy k8s --report summary cluster - not working (node-collector pod in pending state)

### DISCUSSION THREAD

**Discussion by arunabhabanerjee** [aquasecurity/trivy#5610] [Bugs]
### Description

I was running "trivy k8s --report summary cluster" command to get the cluster summary report. However, the "node-collector" pod is unable to schedule due to some wrong scheduling policy.

### Desired Behavior

I should get a summary report.

### Actual Behavior

`➜ trivy k8s --report summary cluster --debug
2023-11-19T13:17:32.733+0530	DEBUG	Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2023-11-19T13:17:32.735+0530	DEBUG	Ignore statuses	{"statuses": null}
2023-11-19T13:22:33.581+0530	FATAL	get k8s artifacts with node info error:
    github.com/aquasecurity/trivy/pkg/k8s/commands.clusterRun
        github.com/aquasecurity/trivy/pkg/k8s/commands/cluster.go:34
  - running node-collector job: runner received timeout
`

### Reproduction Steps

```bash
1. Run "trivy k8s --report summary cluster --debug" command
```

### Target

Kubernetes

### Scanner

Vulnerability

### Output Format

Table

### Mode

Standalone

### Debug Output

```bash
➜ trivy k8s --report summary cluster --debug
2023-11-19T13:17:32.733+0530	DEBUG	Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2023-11-19T13:17:32.735+0530	DEBUG	Ignore statuses	{"statuses": null}
2023-11-19T13:22:33.581+0530	FATAL	get k8s artifacts with node info error:
    github.com/aquasecurity/trivy/pkg/k8s/commands.clusterRun
        github.com/aquasecurity/trivy/pkg/k8s/commands/cluster.go:34
  - running node-collector job: runner received timeout

➜ kc get jobs -A
NAMESPACE      NAME                       COMPLETIONS   DURATION   AGE
trivy-system   kube-bench                 1/1           10s        51m
trivy-temp     node-collector-d667cfb6d   0/1           70s        70s

➜ kc get po -n trivy-temp
NAME                             READY   STATUS    RESTARTS   AGE
node-collector-d667cfb6d-nd5c5   0/1     Pending   0          2m4s

➜ kc describe po node-collector-d667cfb6d-nd5c5 -n trivy-temp
.........
Events:
  Type     Reason            Age    From               Message
  ----     ------            ----   ----               -------
  Warning  FailedScheduling  2m18s  default-scheduler  0/4 nodes are available: 1 node(s) had untolerated taint {node-role.kubernetes.io/control-plane: }, 3 node(s) didn't match Pod's node affinity/selector. preemption: 0/4 nodes are available: 4 Preemption is not helpful for scheduling..

➜ kc get no
NAME           STATUS   ROLES                  AGE   VERSION
k8s-master-0   Ready    control-plane,master   8d    v1.27.3
k8s-worker-0   Ready    worker                 8d    v1.27.3
k8s-worker-1   Ready    worker                 8d    v1.27.3
k8s-worker-2   Ready    worker                 8d    v1.27.3
```

### Operating System

macOS Sonoma

### Version

```bash
➜ trivy --version
Version: 0.47.0
```

### Checklist

- [ ] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by chen-keinan** [comment]
@arunabhabanerjee have you tried to use toleration ?  see [flag options](https://aquasecurity.github.io/trivy/v0.47/docs/references/configuration/cli/trivy_kubernetes/)
